.github/workflows/functional-tests.yml

```diff
@@ -10,6 +10,10 @@ on:
             - '.github/workflows/functional-tests.yml'
             - 'src/Turbo/**'
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
```

Comment:
I've found functional tests were not canceled when pushing a new commit when they are already running.